### PR TITLE
add feature repeat given column when split across pages

### PR DESCRIPTION
**Pull request by singhinderpal**
just pass the data key to repeat in config
`horizontalPageBreakRepeat: 'id'`

### CODE REVIEW

**Comment by singhinderpal** [comment]
#800 Linking to the issue.

File: src/tablePrinter.ts [hunk 1]
```diff
@@ -23,11 +23,28 @@ const getColumnsCanFitInPage = (
   // get page width
   const availablePageWidth = getPageAvailableWidth(doc, table)
   let remainingWidth = availablePageWidth
+  // get column data key to repeat
+  const horizontalPageBreakRepeat = table.settings.horizontalPageBreakRepeat;
+  let repeatColumnIndex = -1;
```
Comment (simonbengtsson) [inline src/tablePrinter.ts:28]: I think we can simplify some code here if we remove the repeatColumnIndex in favour of moving `repeatCol` outside of the if statement. Something like this:

```
const repeatColumn = null
if (horizontalPageBreakRepeat) {
    repeatColumn = table.columns.find((item) => item.dataKey == horizontalPageBreakRepeat);
    // ...
  }
  while (i < len) {
    if (repeatColumn?.index === i) {
    // ...
```

**Comment by simonbengtsson** [comment]
Looks great! Had some minor comments. Also, does this support specifying a column index instead of a key? For example 

`horizontalPageBreakRepeat: 0`

**Review by simonbengtsson — CHANGES_REQUESTED** [review]

**Comment by singhinderpal** [comment]
Hello @simonbengtsson 

The changed have been done and support to consume indexes in horizontalPageBreakRepeat is also added. Kindly review.

**Comment by simonbengtsson** [comment]
Looks great! Two things left:
- Run prettier on it to format it like the rest of the code
- Add documentation

**Comment by singhinderpal** [comment]
Hello @simonbengtsson 

I have added an example for horizontalPageBreakRepeat and ran prettier to format the code,

I request you to review and merge.

**Comment by simonbengtsson** [comment]
No need for an example, but would be great if the option would be described in the documentation (ie README.md).

**Comment by singhinderpal** [comment]
Hi @simonbengtsson 
Its now added under Other options section.

**Comment by simonbengtsson** [comment]
Looks great!

**Pull request merged**